packages/contracts/contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -1733,7 +1723,8 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
     transactionContext.ovmL1TXORIGIN = _transaction.l1TxOrigin;
     transactionContext.ovmGASLIMIT = gasMeterConfig.maxGasPerQueuePerEpoch;
 
-    messageRecord.nuisanceGasLeft = _getNuisanceGasLimit(_transaction.gasLimit);
+    // Initialize this to
+    messageRecord.nuisanceGasLeft = Math.min(_transaction.gasLimit, gasleft());
```

Comment:
Hmmmm maybe easier to just have `nuisanceGasLeft = _transaction.gasLimit`. EM should be enforcing that `gasleft() > _transaction.gasLimit` near the start. Also because I hate `gasleft()` as an opcode and it introduces a lot of annoying complexity into the fraud proof.